python/pfs/pipe2d/generateReductionSpec.py

```diff
@@ -59,6 +61,88 @@ class BeamConfig:
     pfs_design_id: int
 
 
+@dataclasses.dataclass
+class SelectionCriteria:
+    """Selection criteria used in queries to opDB.
+    """
+
+    date_start: Optional[datetime.datetime] = None
+    date_end: Optional[datetime.datetime] = None
+    visit_start: Optional[int] = None
+    visit_end: Optional[int] = None
+
+    @classmethod
+    def fromNamespace(cls, args: argparse.Namespace, *, remove: bool = False) -> "SelectionCriteria":
+        """Construct SelectionCriteria from argparse.Namespace object.
+
+        Parameters
+        ----------
+        args : `argparse.Namespace`
+            A return value of ``argparse.ArgumentParser().parse_args()``.
+        remove : `bool`
+            Remove from ``args`` those members that are used by this function.
+
+        Returns
+        -------
+        criteria : `SelectionCriteria`
+            Constructed SelectionCriteria instance.
+        """
+        criteria = SelectionCriteria()
+        criteria.updateFromNamespace(args, remove=remove)
+        return criteria
+
+    def updateFromNamespace(self, args: argparse.Namespace, *, remove: bool = False):
+        """Update self using members in argparse.Namespace object.
+
+        Parameters
+        ----------
+        args : `argparse.Namespace`
+            A return value of ``argparse.ArgumentParser().parse_args()``.
+        remove : `bool`
+            Remove from ``args`` those members that are used by this function.
+        """
+        undefined = object()
```

Comment:
Why not use `None`?

Reply:
If I use None here, the caller cannot overwrite a field with None. (I will add an explanation to the code).